indico_toolkit/metrics/usermetrics.py

```diff
@@ -0,0 +1,92 @@
+import pandas as pd
+from typing import List, Dict
+from indico import IndicoClient
+from datetime import datetime
+from indico_toolkit.indico_wrapper import IndicoWrapper
+from indico_toolkit import ToolkitInputError
+from indico.queries.usermetrics import GetUserSnapshots
+from indico.filters import UserMetricsFilter
+
+
+class UserMetrics(IndicoWrapper):
+    def __init__(self, client: IndicoClient):
+        self.client = client
+
+    def get_user_metrics_df(
+        self,
+        columns: List[str] = [
+            "id",
+            "name",
+            "email",
+            "enabled",
+            "roles",
+            "created_at",
+            "datasets",
+        ],
+        date: datetime = datetime.now(),
+        filter_user_id: int = None,
+        filter_email: str = None,
+    ) -> pd.DataFrame:
+        """
+        Get a dataframe of user metrics in the caller's cluster
+        Args:
+            columns (List[str]): List of columns to include in the dataframe (OPTIONAL) - defaults to all
+            date (datetime): specific date to query - defaults to current datetime (OPTIONAL)
+            filter_user_id (int): user_id to filter for (OPTIONAL)
+            filter_email (str): user email to filter for (OPTIONAL)
+        """
+        filter_opts = {}
+        if filter_user_id or filter_email:
+            filter_opts = UserMetricsFilter(
+                user_id=filter_user_id, filter_email=filter_email
+            )
+
+        all_user_metrics = []
+        for snapshots in self.client.paginate(
+            GetUserSnapshots(date=date, filters=filter_opts)
+        ):
```

Comment:
This should be refactored to another member function `get_user_snapshots`

Reply:
Resolved here - https://github.com/IndicoDataSolutions/Indico-Solutions-Toolkit/blob/e49e5e55a1c12d9a696330b5f1355051839ed5b2/indico_toolkit/metrics/usermetrics.py#L50